src/Simulation/Simulators/OpenSystemsSimulator/DataModel/ArrayConverter.cs

```diff
@@ -3,249 +3,245 @@
-    ///     and <c>serde</c> crates.
-    ///   </para>
-    ///   <para>
-    ///     In particular, each JSON object produced by this converter has
-    ///     three properties:
-    ///   </para>
-    ///   <list type="bullet">
-    ///     <item><c>"v"</c>: Indicates the <c>ndarray</c> format version being
-    ///       being used. Always <c>1</c> for this implementation.</item>
-    ///     <item><c>"dim"</c>: Lists the dimensions of the array being
-    ///       serialized. Will always contain one less dimension than the
-    ///       original array, as one dimension represents the real and imaginary
-    ///       components of each element.
-    ///     <item><c>"data"</c>: Lists the elements of the array, with the
-    ///       right-most dimension varying the fastest (defined by <c>ndarray</c>
-    ///       as "logical order", and by NumPy as "C-ordered").
-    ///   </list>
-    /// </remarks>
-    public class ComplexArrayConverter : JsonConverter<NDArray>
+    public override NDArray Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        public override NDArray Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        // By the end, we need to have read "v", "dim", and "data".
+        int? version = null;
+        List<int>? dims = null;
+        List<Complex>? data = null;
+
+        // We require that the reader be in the right state to read an
+        // object.
+        if (reader.TokenType != JsonTokenType.StartObject)
         {
-            // By the end, we need to have read "v", "dim", and "data".
-            int? version = null;
-            List<int>? dims = null;
-            List<Complex>? data = null;
-
-            // We require that the reader be in the right state to read an
-            // object.
-            if (reader.TokenType != JsonTokenType.StartObject)
+            throw new JsonException();
+        }
+
+        while (reader.Read())
+        {
+            if (reader.TokenType == JsonTokenType.EndObject)
+            {
+                // We're at the end of the array, and can break out of the
+                // read loop.
+                break;
+            }
+
+            // If it's not the end of the object, the current token needs
+            // to be a property name.
+            if (reader.TokenType != JsonTokenType.PropertyName)
             {
                 throw new JsonException();
             }
 
-            while (reader.Read())
+            var propertyName = reader.GetString();
+
+            switch (propertyName)
             {
-                if (reader.TokenType == JsonTokenType.EndObject)
-                {
-                    // We're at the end of the array, and can break out of the
-                    // read loop.
+                case "v":
+                    reader.Read();
+                    version = reader.GetInt32();
```

Comment:
(Up to you, feel free to ignore. I can be asking a question that may never be applicable)

Earlier in the lines 29 - 30 there is a fragment 
```c#
///     <item><c>"v"</c>: Indicates the <c>ndarray</c> format version being
///       being used. Always <c>1</c> for this implementation.</item>
```
Is there a chance that "v" can be non-integer? E.g. "1.2.3". If yes then what will the (green) line 78 do? Throw a (hardly actionable) [exception](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.getint32?view=net-6.0#exceptions)?

Reply:
In this case, `"v"` should always be an integer, as per the design of the serde-json and ndarray crates for Rust.